src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/GetAzureDeploymentOperationCmdlet.cs

```diff
@@ -0,0 +1,115 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    using System;
+    using System.Management.Automation;
+    using System.Threading.Tasks;
+    using Commands.Common.Authentication.Abstractions;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+    using Microsoft.Azure.Commands.ResourceManager.Common;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Gets the deployment operation.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRmDeploymentOperation"), OutputType(typeof(PSObject))]
+    public class GetAzureDeploymentOperationCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the deployment name parameter.
+        /// </summary>
+        [Alias("Name")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment name.")]
+        [ValidateNotNullOrEmpty]
+        public string DeploymentName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the subscription id parameter.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The subscription to use.")]
+        [ValidateNotNullOrEmpty]
+        public Guid? SubscriptionId { get; set; }
+
+        /// <summary>
+        /// Executes the cmdlet.
+        /// </summary>
+        protected override void OnProcessRecord()
```

Comment:
this is a GET operation. And it doesn't look natural if we have PSDeployment as InputObject while this cmdlet is about deployment operation.

Reply:
You should name the parameter ```Deployment``` (not ```InputObject```) 
This may be very helpful in the following scenario:
```Powershell
Get-AzureRmDeployment | ? {$_.DeploymentName -like "*test*" } | Get-AzureRmDeploymentOperation
```

Reply:
The mentioned scenario works right now, because we have the "DeploymentName" parameter.

Reply:
Please add a piping scenario example to the related help file

Reply:
done